source/draw/RasterMonochrome.ooc

```diff
@@ -126,7 +126,21 @@ RasterMonochrome: class extends RasterPacked {
 		}
 		original apply(f)
 	}
-
+	// get the derivative on small window, windowLocation is window's global location on image, window is left top centered.
+	getFirstDerivativeWindow: func(windowLocation: IntBox2D, imageX, imageY: FloatImage) {
+		step := 3
+		// Ix & Iy  centered difference approximation with 4th error order, centered window
+		for (y in (windowLocation leftTop y)..(windowLocation rightBottom y)) {
+			for (x in (windowLocation leftTop x)..(windowLocation rightBottom x)) {
+				imageX[x - windowLocation leftTop x, y - windowLocation leftTop y] = ((-(this[x + 2 * step, y] y)) as Float + 8.0f * (this[x + step, y] y) as Float - 8.0f * (this[x - step, y] y) as Float + (this[x - 2 * step, y] y) as Float) / (12.0f * step)
```

Comment:
Maybe, but there is also the question of legibility to consider. You also lose accuracy by converting to Floats in several places instead of summing Ints together and casting to an Float at the end, right before the division.